### ISSUE TITLE
The editor should allow to load and edit any text file not only scad files.

### ISSUE THREAD

**Issue by yahbluez** [openscad/openscad#4791]
While openscad did not allow scripting actions,
it is useful to run scripts in other langauages
and there for sometimes minor changes to the scripts or data files is necessary.

Would be nice to stay in the same editor.

solution:
remove the fixed file ending and allow to load any kind of text file.

**Comment by jordanbrown0** [comment]
The standard Windows user interface for this is to have the file-type pulldown (in the bottom right of an Open dialog, above [Open] and [Cancel]) include "All Files (*.*)", as in this snippet from Notepad:

![image](https://github.com/openscad/openscad/assets/31949071/cb7669c5-6f78-4e86-8e0a-d865c00ff251)

**Comment by yahbluez** [comment]
![image](https://github.com/openscad/openscad/assets/47297504/07685b60-ce87-46c1-ad6d-58870eaa584f)

In Linux too, but the selector is fixed to openscad files, i can't click it and choose/edit others.

OpenSCAD  version 2023.10.12
Manjaro Linux

**Comment by jordanbrown0** [comment]
Yes, sorry, I was not saying that OpenSCAD did what you want on Windows.  It is the same on Windows; it only offers *.scad and *.csg.  I was trying to point out how it *should* work.

**Comment by jordanbrown0** [comment]
One complexity:  OpenSCAD has some [existing special handling](https://github.com/openscad/openscad/blob/f602162d26be3ed2a4aa6e8e64ae6c5c66fb73bc/src/gui/MainWindow.cc#L295) of a few file types (stl, obj, 3mf, off, dxf, svg, amf, dat, png, json).  You can't open them using the dialog, but if you drag-and-drop one of them on OpenSCAD, or open them via the command line, the effect is to insert an `import()` or `surface()` referring to the file.

**Comment by yahbluez** [comment]
Yah, that work around is be used, but it would be much more useful to remove the limitation from the file open dialog. This limitations are hand made and not the default for file open dialogs.

**Comment by jordanbrown0** [comment]
Maybe the special handling could be disabled when the file is loaded via the Open dialog.

Or maybe you can't edit those few file types.  (But not being able to edit json would be a nuisance.)

Or maybe we could just get rid of that special handling.

But unfortunately [XKCD 1172](https://xkcd.com/1172/) applies.

**Comment by keithel** [comment]
It is super simple to add the "All files (*.*)" filter to the open file dialog, however, as you've mentioned, some file types are treated special when loaded - even through the openFile utility function (which shows the file dialog).
For the moment I'm going to treat that as a feature to allow you to "open" stl files to add the import line for it, although maybe that should instead be initiated by some other menu item or toolbar button (File->Import perhaps).

I could implement that change right now - I know where in the code this is defined, and what it will do.
What I don't know at the moment is how the special handling is dealt with and how to circumvent it if desired, and instead just read in the file and display it in the editor.

I'm about to introduce a PR that adds an open file dialog filter for STLs, so you can show STLs in the file dialog so the user can easily select them to add import lines to their projects.

**Comment by yahbluez** [comment]
Thank you.
The idea to import the path of a file as a button in the file dialog is a good idea. It belongs there and i see not the need for an additional menu entry.

Being able to open any text file also allows to open often used README files or LOG files which some may use beside of scripts to automate openscad operations.

Would like to see that your PR would accepted.

**Comment by nophead** [comment]
I think you can just drag an STL file to the editor pane to get an import statement for it. Seems more logical as you can chose where it is dropped.

**Comment by keithel** [comment]
@nophead that isn't how drag-import works today.
If you drag-import an STL, and land your mouse pointer at any place in the OpenSCAD window (within or outside the editor portion), it will place the import statement wherever your text cursor is located.

**Comment by jordanbrown0** [comment]
I increasingly think that "add an import statement for this file" should be considered a kind of template, not a kind of open.

**Comment by yahbluez** [comment]
When I issued this, nearly two years ago, one of the solutions I was told made my way. I just stopped using the internal editor at all and moved over using an external editor with a lot of enhancements for openscad. I use microsoft code which uns fine with Linux.

My personal use of the internal editor is now limited to small snipes maybe from reddit or else to help/test/look some code.

It is sometimes hard to understand how developers made decisions.

Here for example the idea to filter the file name pattern to force the use of some file types instead of just using the default which allows open anything. I never expected that this tiny request runs into a discussion.  It would still be great to be able to edit any file with the internal editor, for example bash or python scripts used to automate the run of openscad. Which was why i asked. I used bash two years ago and now make all automations with python.

**Comment by jordanbrown0** [comment]
The expected behavior for Windows applications is that when you ask them to open a file, they offer you files of the type that the program supports.  Allowing you to attempt to open, say, a `.png` file will not make anybody happier.  And for a long time, `.scad` files were the only files that were particularly interesting to open - and mostly, they still are.  Yes, it could have offered the `*.*` option as the second option, but it was never important enough for anybody to bother.  Somewhere in there somebody added the "automatically insert import operations" feature, and now adding the `*.*` operation would interact strangely with that feature, and in particular would still mean that you couldn't open some file types.  Could it be fixed?  Sure.  Is it important enough to anybody to spend the effort figuring out how to reconcile the behaviors?  Apparently not.

**Comment by yahbluez** [comment]
Yah, i know that opinion. But because editing other than scad text files is so important (at least for me) i moved over to a separate editor which solves that and other issues with the build in one.  The development of the openscad core is much more important than a  minor editor enhancement.
